### PR TITLE
Format

### LINKED ISSUE #1496: Replace string .format with f-string
## 🚀 Feature

Once #1495 is merged we can replace all `.format` with f-strings.

If interested in helping with the issue, please comment here and send a PR according to our contributing guide.
Feel free to ask questions for details if needed.

### PR DESCRIPTION
**Pull request by uribgp**
Fixes #1496

Description:
Sorry if this is spamming your repo!  I think I've fixed the last issue for now.

Check list:
* [ ] New tests are added (if a new feature is added)
* [ ] New doc strings: description and/or example code are in RST format
* [ ] Documentation is updated (if required)

### CODE REVIEW

**Comment by vfdev-5** [comment]
@uribgp thanks for the PR ! 

> Sorry if this is spamming your repo! I think I've fixed the last issue for now.

No worries. Yes, it is also possible to make all the changes in a single PR and provide several commits instead of multiple PRs :) 

Currently, the CI for python>=3.6 fails. Please, take a look
- https://github.com/pytorch/ignite/pull/1499/checks?check_run_id=1555537930#step:8:18

Please, commit to the same branch to fix the issue and thus we can avoid multiple PRs for the same issue fix.

File: docs/source/faq.rst [hunk 1]
```diff
@@ -375,11 +375,11 @@ Simpliest way to fetch time of single epoch and complete training is to use
 
     @trainer.on(Events.EPOCH_COMPLETED)
     def log_epoch_time():
-        print("{}: {}".format(trainer.state.epoch, trainer.state.times["EPOCH_COMPLETED"]))
+        print(f"{(trainer.state.epoch}: {trainer.state.times["EPOCH_COMPLETED"]}")
```
Comment (trsvchn) [inline docs/source/faq.rst:378]: Needless parenthesis! `(trainer.state.epoch`

File: docs/source/metrics.rst [hunk 2]
```diff
@@ -66,7 +66,7 @@ use the ``output_transform`` argument to transform it:
 
     .. code-block:: python
 
-        device = "cuda:{}".format(local_rank)
+        device = "cuda:{local_rank}"
```
Comment (trsvchn) [inline docs/source/metrics.rst:69]: This not a f-string, just a string, no `f` at the beginning.

File: docs/source/faq.rst [hunk 3]
```diff
@@ -375,11 +375,11 @@ Simpliest way to fetch time of single epoch and complete training is to use
...
 
     @trainer.on(Events.COMPLETED)
     def log_total_time():
-        print("Total: {}".format(trainer.state.times["COMPLETED"]))
+        print(f"Total: {trainer.state.times["COMPLETED"]}")
```
Comment (trsvchn) [inline docs/source/faq.rst:382]: SyntaxError: `""` inside `""`

File: examples/contrib/cifar10/main.py [hunk 4]
```diff
@@ -251,20 +251,20 @@ def log_metrics(logger, epoch, elapsed, tag, metrics):
 
 
 def log_basic_info(logger, config):
-    logger.info("Train {} on CIFAR10".format(config["model"]))
-    logger.info("- PyTorch version: {}".format(torch.__version__))
-    logger.info("- Ignite version: {}".format(ignite.__version__))
+    logger.info(f"Train {config["model"]} on CIFAR10")
```
Comment (trsvchn) [inline examples/contrib/cifar10/main.py:254]: SyntaxError: `""` inside `""`

File: ignite/contrib/handlers/lr_finder.py [hunk 5]
```diff
@@ -306,20 +306,20 @@ def attach(
 
         if not isinstance(to_save["optimizer"], torch.optim.Optimizer):
             raise TypeError(
-                "Object to_save['optimizer'] should be torch optimizer, but given {}".format(type(to_save["optimizer"]))
+                f"Object to_save['optimizer'] should be torch optimizer, but given {type(to_save["optimizer"])}"
```
Comment (trsvchn) [inline ignite/contrib/handlers/lr_finder.py:309]: SyntaxError: nested `""`

File: ignite/contrib/handlers/neptune_logger.py [hunk 6]
```diff
@@ -511,14 +511,14 @@ def __call__(self, engine: Engine, logger: NeptuneLogger, event_name: Union[str,
             raise TypeError("Handler GradsScalarHandler works only with NeptuneLogger")
 
         global_step = engine.state.get_event_attrib_value(event_name)
-        tag_prefix = "{}/".format(self.tag) if self.tag else ""
+        tag_prefix = "{self.tag}/" if self.tag else ""
```
Comment (trsvchn) [inline ignite/contrib/handlers/neptune_logger.py:514]: not a f-string

File: docs/source/faq.rst [hunk 7]
```diff
@@ -375,11 +375,10 @@ Simpliest way to fetch time of single epoch and complete training is to use
 
     @trainer.on(Events.EPOCH_COMPLETED)
     def log_epoch_time():
-        print(f"{(trainer.state.epoch}: {trainer.state.times["EPOCH_COMPLETED"]}")
-
+        print(f"{trainer.state.epoch}: {trainer.state.times['EPOCH_COMPLETED']}")
```
Comment (vfdev-5) [inline docs/source/faq.rst:378]:
```suggestion
        print(f"{trainer.state.epoch}: {trainer.state.times['EPOCH_COMPLETED']}")

```

File: docs/source/concepts.rst [hunk 8]
```diff
@@ -181,8 +181,7 @@ reference returned by :meth:`~ignite.engine.engine.Engine.add_event_handler`. Th
     evaluator = create_supervised_evaluator(model, metrics={"acc": Accuracy()})
 
     def log_metrics(engine, title):
-        print("Epoch: {} - {} accuracy: {:.2f}"
-               .format(trainer.state.epoch, title, engine.state.metrics["acc"]))
+        print(f"Epoch: {trainer.state.epoch} - {title} accuracy: {engine.state.metrics["acc"]:.2f}")
```
Comment (vfdev-5) [inline docs/source/concepts.rst:184]:
```suggestion
        print(f"Epoch: {trainer.state.epoch} - {title} accuracy: {engine.state.metrics['acc']:.2f}")
```

File: docs/source/concepts.rst [hunk 9]
```diff
@@ -366,7 +364,7 @@ In the code below, `engine.state.output` will be a list of loss, y_pred, y for t
     def print_loss(engine):
         epoch = engine.state.epoch
         loss = engine.state.output[0]
-        print ('Epoch {epoch}: train_loss = {loss}'.format(epoch=epoch, loss=loss))
+        print (f'Epoch {epoch}: train_loss = {loss}')
```
Comment (vfdev-5) [inline docs/source/concepts.rst:367]:
```suggestion
        print (f"Epoch {epoch}: train_loss = {loss}")
```

File: docs/source/concepts.rst [hunk 10]
```diff
@@ -394,7 +392,7 @@ batch, this is how the user can use `output_transform` to get y_pred and y from
     def print_loss(engine):
         epoch = engine.state.epoch
         loss = engine.state.output['loss']
-        print ('Epoch {epoch}: train_loss = {loss}'.format(epoch=epoch, loss=loss))
+        print (f'Epoch {epoch}: train_loss = {loss}')
```
Comment (vfdev-5) [inline docs/source/concepts.rst:395]:
```suggestion
        print (f"Epoch {epoch}: train_loss = {loss}")
```

File: examples/mnist/mnist_with_tensorboard.py [hunk 11]
```diff
@@ -92,9 +92,8 @@ def run(train_batch_size, val_batch_size, epochs, lr, momentum, log_interval, lo
     @trainer.on(Events.ITERATION_COMPLETED(every=log_interval))
     def log_training_loss(engine):
         print(
-            "Epoch[{}] Iteration[{}/{}] Loss: {:.2f}"
-            "".format(engine.state.epoch, engine.state.iteration, len(train_loader), engine.state.output)
-        )
+            f"Epoch[{engine.state.epoch}] Iteration[{engine.state.iteration}/{len(train_loader)}] Loss: {engine.state.output:.2f}"
+            ""
```
Comment (vfdev-5) [inline examples/mnist/mnist_with_tensorboard.py:96]:
```suggestion
```

File: examples/mnist/mnist_with_visdom.py [hunk 12]
```diff
@@ -83,8 +83,8 @@ def run(train_batch_size, val_batch_size, epochs, lr, momentum, log_interval):
     @trainer.on(Events.ITERATION_COMPLETED(every=log_interval))
     def log_training_loss(engine):
         print(
-            "Epoch[{}] Iteration[{}/{}] Loss: {:.2f}"
-            "".format(engine.state.epoch, engine.state.iteration, len(train_loader), engine.state.output)
+            f"Epoch[{engine.state.epoch}] Iteration[{engine.state.iteration}/{len(train_loader)}] Loss: {engine.state.output:.2f}"
+            ""
```
Comment (vfdev-5) [inline examples/mnist/mnist_with_visdom.py:87]:
```suggestion
```

**Review by vfdev-5 — COMMENTED** [review]
Thanks for working on this PR @uribgp !
I left few comments here and just realized that there 122 files modified.
It would be better to split it into several PR to simplify reviewing, for example, by modules:
- ignite.engine
- ignite.metrics
- ...
Could you please split it into 3-4 PRs and we could proceed quickly on that.

File: examples/mnist/mnist_with_visdom.py (hunk 12 above)
Comment (uribgp) [reply examples/mnist/mnist_with_visdom.py:87]: I think I've fixed all those, plus an extra one.  I also began on some others.

**Comment by vfdev-5** [comment]
@uribgp sounds good. There are still few things to do:
- please, update your branch to the master (by merging it, see https://github.com/pytorch/ignite/blob/master/CONTRIBUTING.md#send-a-pr) and fix the conflict
- it would be really helpful to split the PR into several once, such that we could review faster and merge faster.
Thanks

**Comment by uribgp** [comment]
Broken down into smaller pulls

**Pull request closed without merging**